Modules/Core/Transform/include/itkMatrixOffsetTransformBase.hxx

```diff
@@ -104,6 +106,7 @@ MatrixOffsetTransformBase<TParametersValueType, VInputDimension, VOutputDimensio
   m_InverseMatrix.SetIdentity();
   m_InverseMatrixMTime = m_MatrixMTime;
   this->Modified();
+  this->SetFixedParameters(tmp_fixed_parameters);
```

Comment:
For my understanding: Instead of restoring the original fixed parameters, wouldn't it be possible to just prevent them from being modified in the first place? 

Specifically, wouldn't it be sufficient to remove the existing `m_Center.Fill(InputPointValueType{})` call from `MatrixOffsetTransformBase::SetIdentity()`?

https://github.com/InsightSoftwareConsortium/ITK/blob/1b01d5ceb899f19b2ca104249770854dd8244eab/Modules/Core/Transform/include/itkMatrixOffsetTransformBase.hxx#L102

Reply:
@N-Dekker   I tried that at first, but "parameters" are not part of the base class, they are part of the child classes, and some child classes started failing when I tried to do that. (the Scale*Transforms did not behave as expected).

I also tried to take an approach similar to the "GetInverse" function by replacing the m_Matrix with an identity matrix, but again the "Scale*Transforms" tests started failing.

Your suggestion would work for most transforms who's stationary components are the center of rotation, but as an abstract concept a transform may be defined with other fixedparameters other than center of rotation.  I did not want to identify those cases.

This was an attempt to enforce the minimal concepts that "FixedParameters" identify the stationary component of the transform, and should not be changed by setting to identity.

Reply:
Sorry, I still don't fully understand why some (old) unit tests would require changing FixedParameters during the execution of `SetIdentity()`. It seems to me that your proposed fix is in fact a workaround to this behavior. Maybe I can look at it closer later this week, but otherwise I'm at least glad that most of my comments were helpful to you 😃